photutils/psf/tests/test_photometry.py

```diff
@@ -497,3 +498,28 @@ def test_iterative_psf_photometry_inputs():
     with pytest.raises(ValueError, match=match):
         _ = IterativePSFPhotometry(psf_model, fit_shape, finder=finder,
                                    aperture_radius=4, maxiters=3.14)
+
+
+@pytest.mark.skipif(not HAS_SCIPY, reason='scipy is required')
+def test_out_of_bounds_centroids():
+    sources = Table()
+    sources['id'] = np.arange(4) + 1
+    sources['flux'] = 1
+    sources['x_0'] = [-1.4, 34.5, 14.2, -0.7]
+    sources['y_0'] = [13, -0.2, -1.6, 40]
```

Comment:
Can you also add a sources on the top and right edges?  The negative positions would actually not fail for `data[ycen, xcen]` because of negative-index-wraparound (but not what was intended!).